src/sgame/sg_bot_util.cpp

```diff
@@ -245,6 +245,63 @@ static int GetMaxEquipmentCost( gentity_t const* self )
 	return max_value;
 }
 
+// This function returns a percent value indicating how much
+// resupply action would be desirable.
+// The returned value depends on:
+// * distance between the caller and the closest supply building
+// * caller's skill
+//
+// NOTE: the distance modifier is the same as for BotGetHealScore
+//
+// TODO: use a navigation distance
+// TODO: add poison support
+// TODO: in some cases (goon's barbs) the weapon have infiniteAmmo
```

Comment:
Barb support seems useless to add because you can't replenish them by going to a specific spot.

Reply:
Except that one could be willing to go near a very close booster to regen them much faster

Reply:
The main differences between barbs and most human weapons are:

* barbs slowly regenerate without help
* some buildings can drastically accelerate barb regeneration, which is something players use a lot

Reply:
I always considered booster as a kind of armoury for the barb. The in-game behaviour is the same: return to armoury, reload.